main.py

```diff
@@ -10,12 +11,27 @@
 from functools import partial
 import platform
 
-from config import BIGRAMS_CONFIGS, LAYER_1_LETTERS, LAYER_2_LETTERS, LAYER_3_LETTERS, LAYER_4_LETTERS, VAR_LETTERS_L1_L2, MANUALLY_DEFINE_LAYERS, AUTO_LAYER_SWAP_COUNT, AUTO_LAYER_EMPTY_COUNT, AUTO_LAYER_IGNORE, FIXATE_MOST_COMMON_LETTER, FIXATED_LETTERS, NR_OF_LAYERS, NR_OF_BEST_LAYOUTS, PERFORM_GREEDY_OPTIMIZATION, SHOW_DATA, SHOW_GENERAL_STATS, SHOW_TOP_LAYOUTS, TEST_CUSTOM_LAYOUTS, CUSTOM_LAYOUTS, LETTERS_PER_LAYER, DISABLE_UNICODE, DEBUG_MODE, USE_MULTIPROCESSING, FILL_SYMBOL, SCORE_LIST, SCREEN_WIDTH
+from config import BIGRAMS_CONFIGS, LAYER_1_LETTERS, LAYER_2_LETTERS, LAYER_3_LETTERS, LAYER_4_LETTERS, VAR_LETTERS_L1_L2, MANUALLY_DEFINE_LAYERS, AUTO_LAYER_SWAP_COUNT, AUTO_LAYER_EMPTY_COUNT, AUTO_LAYER_IGNORE, FIXATE_MOST_COMMON_LETTER, FIXATED_LETTERS, NR_OF_LAYERS, NR_OF_BEST_LAYOUTS, PERFORM_GREEDY_OPTIMIZATION, SHOW_DATA, SHOW_GENERAL_STATS, SHOW_TOP_LAYOUTS, TEST_CUSTOM_LAYOUTS, CUSTOM_LAYOUTS, LETTERS_PER_LAYER, DISABLE_UNICODE, DEBUG_MODE, USE_MULTIPROCESSING, USE_CFFI, FILL_SYMBOL, SCORE_LIST, SCREEN_WIDTH
 from helper_classes import BigramsConfig, ConfigSpecificResults
 from ui_helpers import *
 
 start_time = time()
 
+if USE_CFFI:
+    try:
+        from cffi._cffi_extension.lib import test_single_layout, get_top_scores
+        from cffi._cffi_extension import ffi
```

Comment:
Which of these `cffi`s? :-) Do you mean like this?

```py
if USE_FFI:
    try:
        from ffi._ffi_extension.lib import ...
```

Good question! I was even wondering if it shouldn't be "C extension" instead of "CFFI extension".

Reply:
I just noticed I've written it the other way around... I meant to write:
`...why not name it "cffi" instead of "ffi"?`

I was referring to the name you assigned to the imported thing (it should be the last line I highlighted).

Reply:
OK, now I understand. I didn't choose this name. It's always `ffi`. I saw this in the documentation, for example here:

> Then, in your main program, you use:
> ```py
> from _example import ffi, lib
> ```

<https://cffi.readthedocs.io/en/latest/overview.html>

Reply:
Ok, then let's stick to their recommended naming scheme :)